src/bpfmap.h

```diff
@@ -86,9 +98,8 @@ inline std::string bpf_map_name(std::string_view bpftrace_map_name)
   return name;
 }
 
-inline std::string bpftrace_map_name(std::string_view bpf_map_name)
+inline std::string bpftrace_map_name(std::string name)
```

Comment:
Let's leave the parameter as a std::string_view.

A `std::string` parameter results in two string copies being made - one to create the argument and a second to return the result. Making the copy explicitly in the function body means the compiler can take advantage of [named return value optimisation](https://en.cppreference.com/w/cpp/language/copy_elision), so there's no second copy to return the result.

A `const std::string &` parameter would also achieve the same result, but I see now that there's a third string copy that I didn't spot and eliminate the last time I worked on this function. `string::substr` makes a copy, but `string_view::substr` doesn't. The code isn't making use of this fact yet, but it could be done in the future.

Going over all of this, I can see how we ended up with so many string copies throughout bpftrace - it's so easy to accidentally copy a string!

Reply:
Agreed - it's something I didn't really consider before. Thanks for pointing this out!